### PR TITLE
chore(test-data): Good Store product type presets

### PR DESCRIPTION
**Pull request by jaikamat**
## Summary

This PR exposes product presets for the Good Store sample data, which can be found [here](https://github.com/frontastic-developers/customer-thegoodstore/blob/master/getting-started/data/product-type.json#L54). 

Additional comments inline.

### CODE REVIEW

File: models/product-type/src/attribute-definition/attribute-definition-draft/presets/empty.spec.ts [hunk 1]
```diff
@@ -13,11 +13,7 @@ it(`should set attributeConstraint, inputHint, and isSearchable to undefined`, (
       en: expect.any(String),
       fr: expect.any(String),
     }),
-    inputTip: expect.objectContaining({
-      de: expect.any(String),
-      en: expect.any(String),
-      fr: expect.any(String),
-    }),
+    inputTip: undefined,
```
Comment (jaikamat) [inline models/product-type/src/attribute-definition/attribute-definition-draft/presets/empty.spec.ts:16]: `inputTip` is [optional](https://docs.commercetools.com/api/projects/productTypes#attributedefinition), so I included it as a possible undefined. @tylermorrisford or @stephsprinkle, do you see any harm in doing so?

File: models/product-type/src/index.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ export * as AttributeEnumTypeDraft from './attribute-enum-type/attribute-enum-ty
 export * as AttributeLocalizedEnumType from './attribute-localized-enum-type';
 export * as AttributeLocalizedEnumTypeDraft from './attribute-localized-enum-type/attribute-localized-enum-type-draft';
 export * as AttributeLocalizedEnumValue from './attribute-localized-enum-value';
+export * as AttributeLocalizedEnumValueDraft from './attribute-localized-enum-value/attribute-localized-enum-value-draft';
```
Comment (jaikamat) [inline models/product-type/src/index.ts:25]: Needed this downstream

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  ProductTypeDraft.presets
+    .empty()
+    .key('bedding-bundle')
+    .name('Bedding Bundle')
+    .description('Product type used to bundle bedding components')
+    .attributes([
+      AttributeDefinitionDraft.presets
+        .empty()
+        .name('product-description')
+        .type(AttributeLocalizableTextType.random())
```
Comment (jaikamat) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts:19]: Majority of the work on all of these was reconciling the `type` from the related data and ensuring it was accurate. `type` was not included in the autogenerated presets

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/product-sets.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    .attributes([
+      AttributeDefinitionDraft.presets
+        .empty()
+        .name('type')
+        .type(
+          AttributeSetTypeDraft.random().elementType(
+            AttributeLocalizableTextType.random()
+          )
+        )
+        .label(LocalizedString.presets.empty()['en-GB']('Type')['de-DE']('Typ'))
```
Comment (jaikamat) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/product-sets.ts:25]: In general, if US is missing in the localization data, do we include it?

File: models/product-type/src/attribute-definition/attribute-definition-draft/presets/empty.spec.ts (hunk 1 above)
Comment (stephsprinkle) [reply models/product-type/src/attribute-definition/attribute-definition-draft/presets/empty.spec.ts:16]: Seems totally reasonable to me!

File: models/product-type/src/index.ts (hunk 2 above)
Comment (stephsprinkle) [reply models/product-type/src/index.ts:25]: oops

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        .label(LocalizedString.presets.empty()['en-GB']('product-description'))
```
Comment (stephsprinkle) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts:20]: We'll need to include `en-US` for all localized string occurrences

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        .inputTip(
+          LocalizedString.presets
+            .empty()
+            ['en-GB']('description of the bundle product')
+        )
+        .isRequired(false)
+        .attributeConstraint('SameForAll')
```
Comment (stephsprinkle) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts:27]: Need to use the `attributeConstraint` constant

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        .isSearchable(false)
+        .inputHint('SingleLine'),
```
Comment (stephsprinkle) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts:29]: Same here, there's an `inputHint` constant

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        .label(LocalizedString.presets.empty()['en-GB']('Product Spec'))
+        .isRequired(false)
+        .attributeConstraint('SameForAll')
+        .isSearchable(false)
+        .inputHint('SingleLine'),
+
+      AttributeDefinitionDraft.presets
+        .empty()
+        .name('product-ref')
+        .type(AttributeLocalizableTextType.random())
```
Comment (stephsprinkle) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts:44]: This is a [product reference type](https://github.com/frontastic-developers/customer-thegoodstore/blob/59b614b30c58410dff5d23bb9e4e6548566a2d4f/getting-started/data/product-type.json#L410-L413)

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/furniture-and-decor.ts [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+          LocalizedString.presets
+            .empty()
+            ['en-GB'](
+              'Details on the specifications of the product, including size, wood type, fabric type, to be detailed in bullet points.'
+            )
+        )
+        .isRequired(false)
+        .attributeConstraint('SameForAll')
+        .isSearchable(false)
+        .inputHint('MultiLine'),
```
Comment (stephsprinkle) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/furniture-and-decor.ts:41]: All of the attribute definitions in this product type have a `displayGroup` [attribute](https://github.com/frontastic-developers/customer-thegoodstore/blob/59b614b30c58410dff5d23bb9e4e6548566a2d4f/getting-started/data/product-type.json#L27)

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/furniture-and-decor.ts [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+        .name('new-arrival')
+        .type(AttributeBooleanTypeDraft.random())
+        .label(
+          LocalizedString.presets
+            .empty()
+            ['en-GB']('New Arrival')
+            ['de-DE']('Neuankömmling')
+        )
+        .isRequired(false)
+        .type(AttributeBooleanTypeDraft.random().name('boolean'))
```
Comment (stephsprinkle) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/furniture-and-decor.ts:108]: Surely we don't need `name` here on the `BooleanTypeDraft`

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/furniture-and-decor.ts [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+          LocalizedString.presets
+            .empty()
+            ['en-GB'](
+              'used to set the colors and labels displayed in the color filter on PLP'
+            )
+        )
+        .isRequired(false)
+        .type(
+          AttributeLocalizedEnumTypeDraft.random().values([
+            AttributeLocalizedEnumValueDraft.random()
```
Comment (stephsprinkle) [inline models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/furniture-and-decor.ts:157]: 🎉

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/product-sets.ts (hunk 4 above)
Comment (stephsprinkle) [reply models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/product-sets.ts:25]: Yes!

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts (hunk 5 above)
Comment (jaikamat) [reply models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts:20]: Do we need `en-DE` for all as well?

Comment (jaikamat) [reply models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/bedding-bundle.ts]: See b2202a33d7e2e7dd06bcb925d78998737df9499a for included US translations

File: models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/furniture-and-decor.ts (hunk 9 above)
Comment (jaikamat) [reply models/product-type/src/product-type/product-type-draft/presets/sample-data-goodstore/furniture-and-decor.ts]: Is that a supported field? I don't see it documented

**Review by tylermorrisford — APPROVED** [review]

**Review by tdeekens — APPROVED** [review]

**Pull request merged**